### ISSUE TITLE
Connecting to multiple Compose containers: Dev container not found.

### ISSUE THREAD

**Issue by chrmarti** [microsoft/vscode-remote-release#6556]
This is a machine translation, so it may be difficult to understand.
I am experiencing the same problem.
A project that was open until the other day has recently stopped opening.

## System

MacBook Pro (13-inch, M1, 2020)
CPU Apple M1
OS 12.3.1（21E258）

## Project

If only one of them is used, the problem will not occur.

```
pj
├── backend  <- open in Container
├── frontend  <- open in Container
├── docker-compose.dev.yml
```

## Problem

The following error occurs

```
[5061 ms] Start: Run in container: for pid in `cd /proc && ls -d [0-9]*`; do { echo $pid ; readlink /proc/$pid/cwd ; readlink /proc/$pid/ns/mnt ; cat /proc/$pid/stat | tr "
[5151 ms] Dev container not found.
[5152 ms] Error: Dev container not found.
[5152 ms]     at Ca (/Users/{user}/.vscode/extensions/ms-vscode-remote.remote-containers-0.231.2/dist/spec-node/devContainersSpecCLI.js:13198:25)
[5152 ms]     at UR (/Users/{user}/.vscode/extensions/ms-vscode-remote.remote-containers-0.231.2/dist/spec-node/devContainersSpecCLI.js:14008:10)
[5153 ms]     at processTicksAndRejections (node:internal/process/task_queues:96:5)
[5154 ms] Exit code 1
[5157 ms] Command failed: /Applications/Visual Studio Code.app/Contents/MacOS/Electron /Users/{user}/.vscode/extensions/ms-vscode-remote.remote-containers-0.231.2/dist/spec-node/devContainersSpecCLI.js run-user-commands --container-data-folder .vscode-server/data/Machine --container-system-data-folder /var/vscode-server --workspace-folder /Users/{user}/Project/

~~

mount-workspace-git-root true
[5157 ms] Exit code 1
```

I have formatted the devContainersSpecCLI.js to make the issue easier to understand

```/Users/{user}/.vscode/extensions/ms-vscode-remote.remote-containers-0.231.2/dist/spec-node/devContainersSpecCLI.js```

```js:devContainersSpecCLI.js
function Ca(t, e) {
  throw t.write(Se(e)), new Error(e) // line 13198
}
var Hw = Z(require("path"));
```

```js:devContainersSpecCLI.js
    let {
      config: $,
      workspaceConfig: re
    } = W, P = await ms(L, A);
    P || Ca(B.output, "Dev container not found.");  // Line 14008
    let x = await vi(L, P.Id, re.workspaceFolder, $.remoteUser),
      R = cc(B, x, $),
      I = await lc(B, x, $, R, p);
    console.log(JSON.stringify({
      outcome: "success",
      result: I
    }))
  } catch (C) {
    console.error(C), await E(), process.exit(1)
  }
  await E(), process.exit(0)
}
```

I solved the problem by lowering the version as well.

![スクリーンショット 2022-04-04 10 32 20](https://user-images.githubusercontent.com/54758260/161477901-763b5837-2d42-4152-b592-62f79242c409.png)

_Originally posted by @Mizunanari in https://github.com/microsoft/vscode-remote-release/issues/6541#issuecomment-1087125859_

**Comment by chrmarti** [comment]
@Mizunanari Are you using two devcontainer.json files? Using two VS Code windows? Or how do you connect to two containers if the Docker Compose setup?

**Comment by Mizunanari** [comment]
@chrmarti 
Two devcontainer.json.
Two vscode windows are used.
Everything that can be described in docker compose is described here, and "dockerComposeFile" and processing after connecting to vscode are described in devcontainer.json.

**Comment by chrmarti** [comment]
Fixed in v0.231.4 and v0.232.4-pre-release.

**Comment by Mizunanari** [comment]
Thank you.
